go-update/src/test/java/org/reactome/release/goupdate/JavaVsPerlComparison.java

```diff
@@ -107,94 +156,123 @@ public int compare(GKInstance o1, GKInstance o2)
 						int i = DBObjectComparer.compareInstances(javaGoInst, perlInst, sb, 0, true);
 						if (i> 0)
 						{
+							diffNameToCount.put(perlInst.getExtendedDisplayName(), i);
 							diffCount.incrementAndGet();
-							mainSB.append("\n***\nFor Java-updated instance \""+javaGoInst.toString()+"\" with accession \""+accession+"\" there are " + i + " differences:\n"+sb.toString());
+							mainSB.append(
+								"\n***\n" +
+									"For Java-updated instance '" + javaGoInst + "' with accession '" +
+									accession + "' " + "there are " + i + " differences:\n" + sb.toString());
 						}
 						else
 						{
+							sameNameToCount.put(perlInst.getExtendedDisplayName(), i);
 							sameCount.incrementAndGet();
 						}
 						Predicate<? super SchemaAttribute> isNameAttribute = p-> {
-								return p.getName().equals("name") || p.getName().equals("_displayName");
-							};
-						
-						// Now, we need to compare referrers, since they might have display name changes. Want to make sure we capture those correctly.
-						if (javaGoInst.getSchemClass().getName().equals(ReactomeJavaConstants.GO_BiologicalProcess))
+							return p.getName().equals("name") || p.getName().equals("_displayName");
+						};
+
+						// Now, we need to compare referrers, since they might have display name changes.
+						// Want to make sure we capture those correctly.
+						if (hasSchemaClass(ReactomeJavaConstants.GO_BiologicalProcess, javaGoInst))
 						{
 							StringBuilder sb1 = new StringBuilder();
-	
+
 							@SuppressWarnings("unchecked")
-							List<GKInstance> javaList = ((List<GKInstance>) javaGoInst.getReferers(ReactomeJavaConstants.goBiologicalProcess));
+							List<GKInstance> javaList =
+								((List<GKInstance>) javaGoInst.getReferers(ReactomeJavaConstants.goBiologicalProcess));
 							@SuppressWarnings("unchecked")
-							List<GKInstance> perlList = ((List<GKInstance>) perlInst.getReferers(ReactomeJavaConstants.goBiologicalProcess));
+							List<GKInstance> perlList =
+								((List<GKInstance>) perlInst.getReferers(ReactomeJavaConstants.goBiologicalProcess));
 							if (javaList != null && perlList != null)
 							{
 								javaList = javaList.stream().sorted(dbIdComparator).collect(Collectors.toList());
 								perlList = perlList.stream().sorted(dbIdComparator).collect(Collectors.toList());
-	
+
 								GKInstance javaReferrer =  javaList.get(0);
 								GKInstance perlReferrer = perlList.get(0);
-	
-								int numDiffs = DBObjectComparer.compareInstances(javaReferrer, perlReferrer, sb1, 0, isNameAttribute, true);
+
+								int numDiffs = DBObjectComparer.compareInstances(
+									javaReferrer, perlReferrer, sb1, 0, isNameAttribute, true
```

Comment:
Similar to https://github.com/reactome/data-release-pipeline/pull/121/files/600bdc5e31fbe03661507750fb03f8b195feabe1..cf7396e26c9ffe9bf9acc8024dbddc376adc01d9#r358270477 -- though this time it is a `0` instead of a `1`. Should we have named variables for this?

Reply:
Done!  Variables at the top and the changes to the method calls towards the bottom: https://github.com/reactome/data-release-pipeline/pull/121/commits/ce8ce4883a21048b4e003618de3da7b8ba3ae538#diff-a1b0b46e16fc2dac4e77db0ea04b1bb7R74